skribenten-web/frontend/src/Brevredigering/ModelEditor/components/ScalarEditor.tsx

```diff
@@ -1,82 +1,189 @@
+  field,
+  submitOnChange,
+}: {
+  field: string;
+  fieldType: TScalar;
+  submitOnChange?: () => void;
+}) => {
   switch (fieldType.kind) {
     case "NUMBER": {
-      return <TextField {...commonTextFieldProperties} step={1} type="number" />;
+      return (
+        <AutoSavingTextField
+          field={field}
+          fieldType={fieldType}
+          onSubmit={submitOnChange}
+          step={1}
+          timeoutTimer={2000}
+          type={"number"}
+        />
+      );
     }
     case "DOUBLE": {
-      return <TextField {...commonTextFieldProperties} step={0.1} type="number" />;
+      return (
+        <AutoSavingTextField
+          field={field}
+          fieldType={fieldType}
+          onSubmit={submitOnChange}
+          step={0.1}
+          timeoutTimer={3000}
+          type="number"
+        />
+      );
     }
     case "STRING": {
-      return <TextField {...commonTextFieldProperties} type="text" />;
+      return (
+        <AutoSavingTextField
+          field={field}
+          fieldType={fieldType}
+          onSubmit={submitOnChange}
+          timeoutTimer={3000}
+          type="text"
+        />
+      );
     }
     case "BOOLEAN": {
       // TODO: reimplement when an example template exists
       return <Checkbox>{convertFieldToReadableLabel(field)}</Checkbox>;
     }
     case "DATE": {
-      return <ControlledDatePicker field={field} />;
+      return <ControlledDatePicker field={field} fieldType={fieldType} onSubmit={submitOnChange} />;
     }
   }
 };
 
-function ControlledDatePicker({ field }: { field: string }) {
-  const { control } = useFormContext();
+/**
+ * Componenten har mulighet til å autolagre endringer i feltet etter en gitt timeout dersom onSubmit sendes med.
+ * Ellers, kan den også brukes som et vanlig tekst felt.
+ */
+const AutoSavingTextField = (props: {
+  field: string;
+  fieldType: TScalar;
+  type: "number" | "text";
+  step?: number;
+  timeoutTimer: number;
+  onSubmit?: () => void;
+}) => {
+  const { register, getFieldState, watch, reset, formState } = useFormContext();
+
+  const registerProperties = register(props.field, { required: props.fieldType.nullable ? false : "Må oppgis" });
+  const fieldState = getFieldState(registerProperties.name, formState);
+  const watchedValue = watch(registerProperties.name);
+
+  /**
+   * useEffekten er brukt kun i forbindelse med autolagring
+   * Merk at noen felter er avhengig av at andre felter er fyllt ut, før vi prøver å gjøre et kall til backend
+   */
+  useEffect(() => {
+    if (fieldState.isDirty && !!watchedValue && props.onSubmit) {
+      const timeout = setTimeout(() => {
+        props.onSubmit!();
+      }, props.timeoutTimer);
+
+      return () => clearTimeout(timeout);
+    }
+  }, [fieldState.isDirty, watchedValue, watch, props.timeoutTimer, props.onSubmit, props.field, reset]);
+
+  const commonTextFieldProperties = {
+    ...registerProperties,
+    autoComplete: "off",
+    error: fieldState.error?.message,
```

Comment:
Vil oppslag av error fungere sånn som dette? Ser at før så var det følgende, som har støtte for nøstet-field-path

`const potentialError = get(errors, field)?.message?.toString();`

Reply:
Ah, det vil kanskje fungere fordi getFieldState tar seg av path-problematikken for oss?

Reply:
yep, error vises som forventet :)